api/v1alpha3/types.go

```diff
@@ -376,6 +376,8 @@ type DataDisk struct {
 	NameSuffix string `json:"nameSuffix"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// Managed disk option for a VM
+	ManagedDisk ManagedDisk `json:"managedDisk"`
```

Comment:
let's make this field optional (`// +optional`)  and `omitempty` so it is not breaking. By default, we want to keep the existing behavior.